skimage/feature/_basic_features.py

```diff
@@ -0,0 +1,186 @@
+    sigmas = np.logspace(
+        np.log2(sigma_min),
+        np.log2(sigma_max),
+        num=int(np.log2(sigma_max) - np.log2(sigma_min) + 1),
+        base=2,
+        endpoint=True,
+    )[::-1]
+    if has_dask:
+        out_sigmas = [
+            dask.delayed(_singlescale_basic_features_singlechannel)(
+                img, s, intensity=intensity, edges=edges, texture=texture
+            )
+            for s in sigmas
+        ]
+        features = itertools.chain.from_iterable(
+            dask.compute(*out_sigmas, num_workers=num_workers)
+        )
+    else:
+        with ThreadPoolExecutor(max_workers=num_workers) as ex:
+            out_sigmas = list(
+                ex.map(
+                    lambda s: _singlescale_basic_features_singlechannel(
+                        img, s, intensity=intensity, edges=edges, texture=texture
+                    ),
+                    sigmas,
+                )
+            )
+        features = itertools.chain.from_iterable(out_sigmas)
+    return features
+
+
+def multiscale_basic_features(
+    image,
+    multichannel=True,
+    intensity=True,
+    edges=True,
+    texture=True,
+    sigma_min=0.5,
+    sigma_max=16,
+    num_workers=None,
+):
+    """Local features for a single- or multi-channel nd image.
+
+    Intensity, gradient intensity and local structure are computed at
+    different scales thanks to Gaussian blurring.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image, which can be grayscale or multichannel.
+    multichannel : bool, default False
+        True if the last dimension corresponds to color channels.
+    intensity : bool, default True
+        If True, pixel intensities averaged over the different scales
+        are added to the feature set.
+    edges : bool, default True
+        If True, intensities of local gradients averaged over the different
+        scales are added to the feature set.
+    texture : bool, default True
+        If True, eigenvalues of the Hessian matrix after Gaussian blurring
+        at different scales are added to the feature set.
+    sigma_min : float, optional
+        Smallest value of the Gaussian kernel used to average local
+        neighbourhoods before extracting features.
+    sigma_max : float, optional
+        Largest value of the Gaussian kernel used to average local
+        neighbourhoods before extracting features.
+    num_workers : int or None, optional
+        The number of parallel threads to use. If set to ``None``, the full
+        set of available cores are used
+
+
+    Returns
+    -------
+    features : np.ndarray
+        Array of shape ``(n_features,) + image.shape``
+    """
+    if not any([intensity, edges, texture]):
+        raise ValueError(
+                "At least one of ``intensity``, ``edges`` or ``textures``"
+                "must be True for features to be computed."
+                )
+    if image.ndim >= 3 and multichannel:
```

Comment:
I'd suggest a restructure, `if not multichannel: image = image[..., np.newaxis]`, then have a single unified code path.

Reply:
done!